cms/server/ContestWebServer.py

```diff
@@ -1061,10 +1064,12 @@ def which_language(user_filename):
                              if it is not a recognized language.
 
             """
-            for source_ext, language in SOURCE_EXT_TO_LANGUAGE_MAP.iteritems():
-                if user_filename.endswith(source_ext):
-                    return language
-            return None
+
+            languageSubmitted = self.get_argument("submission_lang")
```

Comment:
Oh huh, MissingArgumentError subclasses 400. That seems a little more legit, fair enough.